python/lzreposync/src/lzreposync/primary_parser.py

```diff
@@ -50,61 +93,202 @@ def get_text(node):
+                    and node.namespaceURI == COMMON_NS
+                    and node.tagName == "package"
+                ):
+                    # New package
+                    doc.expandNode(node)
+                    self.current_package = {}
+                    self.current_hdr = {}
+
+                    ### SETTING FAKE DATA FOR SOME ATTRIBUTES
+                    # TODO: Fix these fake attributes
+                    self.current_hdr["rpmversion"] = "1"
+                    self.current_hdr["size"] = 10000
+                    self.current_hdr["payloadformat"] = "cpio"
+                    self.current_hdr["cookie"] = "cookie_test"
+                    self.current_hdr["sigsize"] = 10000
+                    self.current_hdr["sigmd5"] = "sigmd5_test"
+                    # Setting possibly missing attributes: checking for all dependencies
+                    # importLib doesn't accept None values but rather empty arrays ([])
+                    # TODO: Double check the following list, it might not be completely correct,
+                    #  because, the hdr can have two type of keys referring to the same element (for example: '1156' and 'suggestsname' refer to the same thing),
+                    #  so the list should contain only keys that we looked for during and the parsing, and the missing ones,
+                    #  which means, for example, we shouldn't look for possibly missing '1156' key while we've already
+                    #  set the 'suggestsname' key
+                    possibly_missing_dependencies = [
+                        "provides",
+                        "provideversion",
+                        "provideflags",
+                        "requirename",
+                        "requireversion",
+                        "requireflags",
+                        "changelogname",
+                        "changelogtext",
+                        "changelogtime",
+                        "obsoletename",
+                        "obsoleteversion",
+                        "obsoleteflags",
+                        "conflictname",
+                        "conflictversion",
+                        "conflictflags",
+                        1159,
+                        1160,
+                        1161,
+                        1156,
+                        1157,
+                        1158,
+                        5052,
+                        5053,
+                        5054,
+                        5055,
+                        5056,
+                        5057,
+                        5049,
+                        5050,
+                        5051,
+                        5046,
+                        5047,
+                        5048,
+                    ]
+                    for dep in possibly_missing_dependencies:
+                        if not self.current_hdr.get(dep):
+                            self.current_hdr[dep] = []
```

Comment:
I'm not quite sure what this means..

Let's say that the `primary.xml` file we've parsed didn't contain any `<provides>` entry for some package.

 In that case, the `current_hdr` dict will not have a key "provides" in it.

And `importLib` will throw an error like: `header has no key "provides"` (not exactly but something like that).

That's why I tried to set the missing keys to empty lists.

Reply:
I marked this as a nitpick because it's not something that's important here. My comment focuses on correctness without taking the context into account. You're right that the way you wrote it will work fine in this context. I just wanted to highlight why this construct can behave differently from what one might expect.

Setting missing keys to empty lists is good here, I didn't want to say it isn't. Let me try to explain my point with more details:

`dict.get(key, default=None)` returns either the value associated with `key`, or, if the key is not known, the default value (which itself defaults to `None`). Using `if not` on `dict.get()` evaluates to `True` in two cases:
- key does not exist
- value is faslesy (None, False, empty str, list, tuple etc)

Here, the result of that evaluation only assigns a new empty list. That's an operation that has little risk of breaking something and as such we could everything as you have it.

That said, I would propose this slightly different way. This way covers your intentions a bit better because there is only one case where the evaluation is `True`, and that's the case you explained in your comment:

```python
if dep not in current_hdr:
    self.current_hdr[dep] = []
```

My first comment wasn't really clear about that, sorry for the confusion.